internal/bzlmod/go_deps.bzl

```diff
@@ -44,6 +44,36 @@ the required directives to the "default_gazelle_overrides.bzl" file at \
 https://github.com/bazelbuild/bazel-gazelle/tree/master/internal/bzlmod/default_gazelle_overrides.bzl.
 """
 
+_GAZELLE_ATTRS = {
+    "build_file_generation": attr.string(
```

Comment:
We can do this by just switching this `auto` to `on` in the `_GAZELLE_ATTRS`, which will only apply when a user supplies a `gazelle_override` or `gazelle_default_attributes`, otherwise it will be `auto`.

Reply:
Let me know if you think this `dic` makes sense,